### PR TITLE
Finishing up CVscores visualizer

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This PR aims to complete the tasks in #464 to finalize @pdamodaran's new (and most useful!) `CVScores` visualizer. Here's what's done:

- Modified the legend that labels the average cross-validation score dotted line with the numeric value of `self.cv_scores_mean_`
- Removed the ylim constraint so that `CVScores` plots will be more easily comparable across different models.
- Renames the module from `cv` to `cross_validation` for readability and accessibility to newcomers.
- Some additional cleanup in the docstrings for nice rendering in the documentation.
- Updates the docs with the new names and new plots with unadjusted ylim and updated legend.
- Adds some preliminary test coverage, all of which pass on my os ;)

Still to do:

- [x] See which tests fail on Windows or Python 2.7, and update/annotate as necessary.
- [x] Add more tests! For instance, I'd like to add a test that demonstrates that the visualizer works with sklearn estimators that already have cross validation (e.g. both `Ridge` and `RidgeCV`).

Additional suggestions welcome!

### CODE REVIEW

**Comment by lwgray** [comment]
I also clone your branch and was successful in exploring the visualizer with the new enhancements.

File: docs/api/model_selection/cross_validation.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python3
-# cv.py
+# cross_validation.py
```
[the remark this answers is not in the document]
Comment (rebeccabilbro) [reply docs/api/model_selection/cross_validation.py]: Thanks!

File: yellowbrick/model_selection/cross_validation.py [hunk 2]
```diff
@@ -52,15 +53,14 @@ class CVScores(ModelVisualizer):
         - An object to be used as a cross-validation generator.
         - An iterable yielding train/test splits.
 
-        see the scikit-learn
-        `cross-validation guide <http://scikit-learn.org/stable/modules/cross_validation.html>`_
+        See the scikit-learn `cross-validation guide <https://goo.gl/FS3VU6>`_
```
[the remark this answers is not in the document]
Comment (rebeccabilbro) [reply yellowbrick/model_selection/cross_validation.py]: Oh, good - thank you!

**Comment by rebeccabilbro** [comment]
Ok @lwgray - looks like the tests are passing now! Let me know if there's anything else you think is needed, else please merge when ready!

**Pull request merged**